### DISCUSSION TITLE
Wrap `ast.Expr` into a valid `ast.File`?

### DISCUSSION THREAD

**Discussion by shykes** [cuelang/cue#444] [General] [answered]
I'm playing with the Cue `parser` and `ast` packages. Note: the package docs are very helpful, thank you.

I am trying to wrap a `ast.Expr` into a valid `ast.File`. My definition of "valid" is that I can append the file to `build.Instance.Files`, and the build will correctly pick it up. But currently it does not: all other files are correctly used in the build, except my little frankenstein wrapper...

Here is the relevant wrapping code. Can anyone point me to a mistake I'm making?

```
import (
    cueAst "cuelang.org/go/cue/ast"
)

func exprToFile (filename string, expr cueAst.Expr) (*cueAst.File) {
    return &cueAst.File{
        Filename: filename,
        Decls: []cueAst.Decl{expr},
    }
}
```

Not sure it's relevant, but in my case `expr` is generated by `cuelang.org/go/encoding/json.Extract`. The original json contents is `{"hello": "world"}`.

**Comment by shykes** [comment, marked as answer]
I got the answer from @mpvl on slack: `"cuelang.org/go/cue/ast/astutil".ToFile`. Thank you!

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/444.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
